api/routes/utils.js

```diff
@@ -485,42 +484,50 @@ var utils = {
       }
 
       amount *= -1;
+
+      // Dividends pay 10%
+      var dividendsPercentage = 0.1;
+      var newInvestorReps = portfolio[index].reps + amount;
       var investment = portfolio[index].investments[j];
-      var prevAmount = investment.amount;
 
-      // Revenue is the fraction sold times the valuation
-      var revenue = Math.floor(amount / prevAmount * investment.valuation);
-      var roi = (revenue - amount)/amount;
+      var prevInvestmentAmount = investment.amount;
+      var prevInvestmentPercentage = investment.percentage;
+      var newInvestmentAmount = prevInvestmentAmount - amount;
+
+      // newPercentage / newAmount = oldPercentage / oldAmount (Proportional)
+      var newInvestmentPercentage = Math.floor((newInvestmentAmount * investment.percentage) / prevInvestmentAmount);
 
-      // Adjust the investor's reps
-      portfolio[index].reps += revenue;
+      // New dividends equal old dividends reduced by the difference between
+      // what the investment contributed previously and what it contributes now
+      var dividendsChange = (
+        ((prevInvestmentPercentage * prevInvestmentAmount) -
+        (newInvestmentPercentage * newInvestmentAmount)) * dividendsPercentage
+      );
 
-      // Update the amount
-      investment.amount -= amount;
+      var newInvestorDividends = portfolio[index].dividends - dividendsChange;
 
       // If the amount is now zero, remove the investment
-      if (investment.amount === 0) {
+      if (newInvestmentAmount === 0) {
         portfolio[index].investments.splice(j, 1);
         return portfolio;
       }
 
-      // OldTotal/OldPercentage = newTotal/newPercentage (Proportional)
-      investment.percentage = Math.floor((investment.amount * investment.percentage) / prevAmount);
-
-      // New valuation is the proportion of the old valuation relative to the proportion of the investment remaining
-      investment.valuation = investment.valuation * investment.amount / prevAmount;
-
       // Update the date
       investment.timeStamp = Date.now();
 
-      portfolio[index].investments[j] = investment;
-      portfolio[index].roi = this.updateROI(portfolio[index].roi, amount * roi);
+      // Update the investment's amount and percentage
+      investment.amount = newInvestmentAmount;
+      investment.precentage = newInvestmentPercentage;
+
+      // Adjust the investor's reps and dividends
+      portfolio[index].reps = newInvestorReps;
+      portfolio[index].dividends = newInvestorDividends;
     }
     return portfolio;
   },
 
   // Given a transaction, update all of the percentages for investors
-  updateInvestorPercentagesAndValuations: function(investors, expertCategoryTotal, category, username) {
+  updateInvestorPercentages: function(investors, expertCategoryTotal, category, username) {
```

Comment:
Didn't we want to call this Percentiles to be less confusing?

Percentage: The percentage of an expert category you owned at the time of an investment
Percentile: Your rank relative to other investors/experts on a scale of 0 to 100